portfolio/src/main/webapp/script.js

```diff
@@ -160,17 +165,71 @@ function deleteComments() {
   fetch(request).then(response => getComments());
 }
 
+/** Fetches login and Blobstore. */
+function fetchHome() {
+  fetchLogin();
+  fetchBlobstoreUrl();
+}
+
 /**
- * Fetch the Blobstore URL and append it to the comments form
+ * Fetches the Blobstore URL and append it to the comments form
  * in the action attribute.
  */
 function fetchBlobstoreUrl() {
   fetch('/blobstore-upload-url')
-      .then((response) => {
-        return response.text();
-      })
-      .then((imageUploadUrl) => {
+      .then(response => response.text())
+      .then(imageUploadUrl => {
         const messageForm = document.getElementById('comments-form');
         messageForm.action = imageUploadUrl;
-      })
+      });
+}
+
+/**
+ * Fetches the Login Servlet and displayes a message
+ * according to the condition of the user being logged in or out.
+ */
+function fetchLogin() {
+  fetch('/login').then(response => response.json()).then(user => {
+    if (user.loggedIn == true) {
+      // Create element that welcomes the user and prompts him to log out.
```

Comment:
In this comment and in the one below, you can write the full sentence so it's easy to parse (so here, 'Hello, <user>! Click here to log out.')? Because you construct part of it in the anchor element, it's not immediately clear what it will say on the webpage.

Reply:
Done!